File: spark/pyspark/word_count.py
from pyspark.sql import SparkSession

if __name__ == "__main__":
    # Create SparkSession
    spark = SparkSession.builder \
        .appName("WordCountPython") \
        .getOrCreate()

    # Read input text file
    input_path = "input.txt"
    lines = spark.read.text(input_path)

    # Perform Word Count
    word_counts = lines \
        .selectExpr("explode(split(value, '\\s+')) as word") \
        .groupBy("word") \
        .count()

    # Display word counts
    word_counts.show()

    # Stop SparkSession
    spark.stop()

File: spark/scala/WordCount.scala
import org.apache.spark.sql.SparkSession

object WordCount{
  def main(args: Array[String]): Unit = {
    // Create SparkSession
    val spark = SparkSession.builder()
      .appName("WordCountScala")
      .master("local[*]")
      .getOrCreate()

    // Read input text file
    val inputPath = "input.txt"
    val lines = spark.read.textFile(inputPath)

    // Perform Word Count
    val wordCounts = lines
      .flatMap(_.split("\\s+"))
      .map(word => (word.toLowerCase, 1))
      .reduceByKey(_ + _)

    // Display word counts
    wordCounts.collect().foreach(println)

    // Stop SparkSession
    spark.stop()
  }
}
